### ISSUE TITLE
Deep linking to runs on schedule pages

### ISSUE THREAD

**Issue by Planks** [esamarathon/oengus-webapp#31]

**Comment by BobChao87** [comment]
We could also do `#next`, I can see that being useful for people. We also have the option of using the run's position in the marathon instead of the UUID, though I'm unsure if that's immutable or not.

**Comment by Planks** [comment]
> We could also do `#next`, I can see that being useful for people. We also have the option of using the run's position in the marathon instead of the UUID, though I'm unsure if that's immutable or not.

Good point - I agree we should also have `#next`.  In v1 we already have "now" and "next" highlight boxes at the top of each live schedule, but you can't click on them. So these wouldn't need extra share buttons, you'd just make the boxes hotspots.

**Comment by Planks** [comment]
> We also have the option of using the run's position in the marathon instead of the UUID, though I'm unsure if that's immutable or not.

I don't think linking by any sort of relative ordering is useful because organisers move around runs so much. The use case for deep linking a row is mostly so people can link directly to a specific run in the schedule and we'd want that link to remain valid through all schedule edits.

**Comment by BobChao87** [comment]
> > We also have the option of using the run's position in the marathon instead of the UUID, though I'm unsure if that's immutable or not.
> 
> I don't think linking by any sort of relative ordering is useful because organisers move around runs so much. The use case for deep linking a row is mostly so people can link directly to a specific run in the schedule and we'd want that link to remain valid through all schedule edits.

I was afraid that was the case. Oh well, it was worth a shot as then the link would have been slightly more meaningful. Consistency is better.

**Comment by Planks** [comment]
If @duncte123 would be able to tweak the database, at submission time a "friendly uuid" could be generated and added to the submission row like

`BobChao87-MickeyMouseGame-kasdjf` where kasdjf is just some garbage to deduplicate

Seems a lot of work for a relatively minor feature but it would make the URLs much, much nicer particularly when linked on an instant messenger like Discord.

We don't want the URL to expire if any of the run properties are changed so this is something that would need to be locked in permanently when the submission is first created.

**Comment by BobChao87** [comment]
Yeah, I agree. We'll just stick with our existing immutable properties and maybe in the future we can revisit. I made them a little clearer by making the anchor be `#run-UUID`.

**Comment by BobChao87** [comment]
Will be considered resolved after this issue is fixed. https://github.com/esamarathon/oengus-webapp/issues/32#issuecomment-927260694
